modules/dxp/apps/app-builder-workflow/app-builder-workflow-web/src/main/resources/META-INF/resources/js/pages/apps/ListApps.es.js

```diff
@@ -131,39 +147,52 @@ export default ({history, scope, ...props}) => {
 		});
 	};
 
-	function validateFormViewMissingRequiredFields(app) {
-		const isFormViewMissingRequiredFields = (app) => {
+	const onCloseMissingRequiredFieldsModal = () =>
+		setMissingRequiredFieldsVisible(false);
+
+	function validateMissingRequiredFieldsModal(app) {
+		const formViewMissingRequiredFields = (app) => {
 			return getDataDefinition(app.dataDefinitionId).then(
 				(dataDefinition) => {
 					setCurrentApp(app);
 
-					return getFormViews(
-						dataDefinition.id,
-						dataDefinition.defaultLanguageId
-					).then((formViews) => {
-						formViews = checkRequiredFields(
-							formViews,
+					return getItem(
+						`/o/data-engine/v2.0/data-layouts/${app.dataLayoutId}`
+					).then((formView) => {
+						formView.fields = getFormViewFields(formView);
+
+						formView = checkRequiredFields(
+							[formView],
 							dataDefinition
-						);
+						)[0];
 
-						return formViews.find(({id}) => id === app.dataLayoutId)
-							.missingRequiredFields?.customField;
+						return formView.missingRequiredFields;
 					});
 				}
 			);
 		};
 
-		return isFormViewMissingRequiredFields(app).then((missing) => {
-			if (missing) {
-				return new Promise((resolve, reject) => {
+		return formViewMissingRequiredFields(app).then((missing) => {
+			const getRequiredFieldsModal = (fieldType) =>
```

Comment:
What do you think about rename this function to `openMissingRequiredFieldsModal` ?